### DISCUSSION TITLE
What does it take to use xcode

### DISCUSSION THREAD

**Discussion by ghost** [ravynsoft/ravynos#97] [Q&A] [answered]
I realize I cannot use macos binaries however, can you virtualize the docker version of xcode (also on github)? 
I ask since I see no reason a bsd cannot use docker's xcode.

**Comment by mszoek** [comment, marked as answer]
If you are looking at running something like [Docker-OSX](https://github.com/sickcodes/Docker-OSX), that's a full-on macOS image running Xcode inside a Docker container. I assume it would work if you can run Docker, but the output of Xcode will not be compatible with Airyx - it will build normal macOS binaries.

You could use Xcode to edit code and build UIs (NIBs) which can then be compiled on Airyx. It would be great to find or write some kind of tool that translates Xcode project files into BSD Makefiles to make this easier.

**Reply by seonar22** [reply]
I would like to ask if I can run Xcode downloaded from Apple Developer website and run on airyx? According to Readme, my understanding is, I can only compile macOS applications from source and run on airyx. 

Sorry for butting in another person's thread.

**Reply by mszoek** [reply]
Hi! Your understanding is correct... currently you can design apps with Xcode, copy the source & NIBs to airyxOS, and compile/run the result there. Only fairly simple things work so far - our Cocoa libraries are not up to date and many things are missing. There's still a lot of work to be done here.

I am also working towards MachO compatibility which would let Mac apps run as well, but that's a big task and will not be ready any time soon.

Help is welcome on all areas :)
